src/components/terms-and-conditions.tsx

```diff
@@ -0,0 +1,195 @@
+// Terms and Conditions Pop-up
+
+import { Fragment, useState } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { Controller, useForm } from "react-hook-form";
+import { AiFillCloseCircle } from "react-icons/ai";
+
+import Button from "@/components/Button";
+
+type FormValues = {
+  tncAccepted: boolean;
+};
+
+export const TermsAndConditions = () => {
+  const { control, register } = useForm<FormValues>({
+    defaultValues: {
+      tncAccepted: false
+    }
+  });
+
+  const [showPopup, setShowPopup] = useState(false);
+  const handleshowPopupChange = () => {
+    setShowPopup((prevState) => !prevState);
+  };
+
+  const [Accepted, setAccepted] = useState(false);
+  const handleAcceptedChange = () => {
+    setAccepted((prevState) => !prevState);
+  };
+
+  return (
+    <>
+      <div className="static flex justify-center">
+        <Controller
+          control={control}
+          name="tncAccepted"
+          render={({ field: { onChange } }) => (
+            <input
+              type="checkbox"
+              checked={Accepted}
+              {...register("tncAccepted", {
+                required: "*Accept terms and conditions to submit."
+              })}
+              onChange={(e) => {
+                onChange(e.target.checked);
+              }}
+            />
+          )}
+        />
+
+        <span className="pl-2">I have read and accept the</span>
+        <button
+          type="button"
+          onClick={handleshowPopupChange}
+          className="pl-1.5 text-darkAqua-400 hover:text-darkAqua-200"
+        >
+          house rules
+        </button>
+        <span>.</span>
```

Comment:
Maybe it is just me but the space between `I have read and accept the` and `house rules.` is slightly wider than the rest.